FEXCore/Source/Interface/Core/JIT/Arm64/VectorOps.cpp

```diff
@@ -2204,14 +2208,18 @@ DEF_OP(VSMin) {
   if (HostSupportsSVE256 && Is256Bit) {
     const auto Pred = PRED_TMP_32B.Merging();
 
+    // In any case where the destination aliases one of the source vectors
+    // then we can just perform the SMIN in place.
     if (Dst == Vector1) {
-      // Trivial case where we can perform the operation in place.
       smin(SubRegSize, Dst.Z(), Pred, Dst.Z(), Vector2.Z());
+    } else if (Dst == Vector2) {
+      smin(SubRegSize, Dst.Z(), Pred, Dst.Z(), Vector1.Z());
     } else {
-      // SVE SMIN is a destructive operation, so we need a temporary.
-      movprfx(VTMP1.Z(), Vector1.Z());
-      smin(SubRegSize, VTMP1.Z(), Pred, VTMP1.Z(), Vector2.Z());
-      mov(Dst.Z(), VTMP1.Z());
+      // SVE SMIN is a destructive operation, but we know nothing is
+      // aliasing the destination by this point, so we can move into
+      // the destination without needing a temporary.
+      mov(Dst.Z(), Vector1.Z());
```

Comment:
Do we still want to use movprfx on these so we get the benefit of the instruction fusing to the next one?

Reply:
Oh, yes we do. Dunno why I thought it wasn't necessary anymore.